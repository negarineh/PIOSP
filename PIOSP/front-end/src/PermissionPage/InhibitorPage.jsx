import React from 'react';
import '../components/stylesheets/MainPage.css';
import '../components/stylesheets/Survey.css';
import '../components/stylesheets/Butterfly.css';
import '../components/stylesheets/Permission.css';

import { Segment } from 'semantic-ui-react';

// If participant attended the survey before this component will render
// It show message to the participant for their previous attend and 
// they can not go to the survey questions page
class InhibitorPage extends React.Component {

    render() {
     
      return (
        <div className="mainpage-permission">
            <div className="butterfly">
              <div className="body">
                <img src="http://fjordstudio.dk/animations/img/butterfly3-body.png" alt=""/>
              </div>
              <div className="leftwing">
                  <img src="http://fjordstudio.dk/animations/img/butterfly3-leftwing.png" alt=""/>
              </div>
              <div className="rightwing">
                  <img src="http://fjordstudio.dk/animations/img/butterfly3-rightwing.png" alt=""/>
              </div>
            </div>

          <div className="wrappermainpage-permission">    
                <Segment style={{height:'500px'}}>
                <p key="a" className="pwelcome1" > 
                  You have opened this link before
                </p>
                </Segment>
          </div>
        </div>
      );
    }
  }

export  {InhibitorPage};