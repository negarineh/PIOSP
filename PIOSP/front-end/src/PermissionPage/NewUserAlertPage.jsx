import React from 'react';
import '../components/stylesheets/MainPage.css';
import '../components/stylesheets/Survey.css';
import '../components/stylesheets/Butterfly.css';
import '../components/stylesheets/Permission.css';

// render when someone without creating account wants to access survey page url
class NewUserAlertPage extends React.Component {

    render() {

      return (
        <div className="mainpage-permission">
          <div className="butterfly">
            <div className="body">
              <img src="http://fjordstudio.dk/animations/img/butterfly3-body.png" alt=""/>
            </div>
            <div className="leftwing">
                <img src="http://fjordstudio.dk/animations/img/butterfly3-leftwing.png" alt=""/>
            </div>
            <div className="rightwing">
                <img src="http://fjordstudio.dk/animations/img/butterfly3-rightwing.png" alt=""/>
            </div>
          </div>

          <div className="wrappermainpage-permission">    
            <fieldset  className="fieldsetwelcome1">
              <p key="a" className="pwelcome1" > 
              Sorry Your Account Does Not Exits ,Please Create An Account
              </p>
            </fieldset> 
          </div>
        </div>
      );
    }
  }

export  {NewUserAlertPage};