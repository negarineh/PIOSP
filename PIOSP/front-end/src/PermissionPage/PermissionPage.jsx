import React from 'react';
import '../components/stylesheets/MainPage.css';
import '../components/stylesheets/Survey.css';
import '../components/stylesheets/Butterfly.css';
import '../components/stylesheets/Permission.css';
import { Segment} from 'semantic-ui-react'; 

// render when someone wants to have access to survey pages again
// this will prevent attend agin or manipulate pages 
class PermissionPage extends React.Component {
    
    render() {
     
      return (
        <div className="mainpage-permission">
          {/* <Header/> */}
          <div className="butterfly">
            <div className="body">
              <img src="http://fjordstudio.dk/animations/img/butterfly3-body.png" alt=""/>
            </div>
            <div className="leftwing">
                <img src="http://fjordstudio.dk/animations/img/butterfly3-leftwing.png" alt=""/>
            </div>
            <div className="rightwing">
                <img src="http://fjordstudio.dk/animations/img/butterfly3-rightwing.png" alt=""/>
            </div>
          </div>

          <div className="wrappermainpage-permission">    
            <Segment style={{paddingLeft:"10px"}}>
                  <p key="a" className="pwelcome1" > 
                  We study community ecology in agroecosystems
                  </p>
                  <p key="a" className="pwelcome2" > 
                  1.  Crop Pollinator Identity and Ecology
                  </p>
                  <p key="a" className="pwelcome3" > 
                  Pollinator communities include native and introduced bees, flies, beetles, moths, butterflies, and birds, as well as other taxa. These taxa provide pollination services for up to 87.5% of all flowering plants and 75% of the leading global food crops. We are interested in finding out which pollinators visit which crops in a range of agroecosystems. This project is supported by funding from the Australian Government Department of Agriculture and Water resources as part of its Rural R&D for Profit Programme and the Ian Potter Foundation.
                  </p>
                  <p key="a" className="pwelcome2" > 
                  2.  The Mechanisms Underlying Crop Pollinator Effectiveness
                  </p>
                  <p key="a" className="pwelcome3" > 
                  Maximally effective pollination results from a complex assortment of factors that influence the pollination process before and after pollen deposition. The effectiveness of individual pollinators is determined by the combined outcomes of the amount, quality and timing of the pollen transferred which is strongly related to pollinator foraging behaviour, morphology and density. Variation in the identity and spatial and temporal structuring of plant and pollinator communities further affect effective pollination, impacting pollen availability and post-pollination factors, such as pollen-pistil and/or pollen-pollen interactions on the stigma. Pollination failure can thus result from problems at any or all of these stages of pollen transfer. This project is funded by the Australian Research Council via a Discovery Early Career Researcher Award (DECRA) awarded to Romina Rader (DE170101349).
                  </p>
                  <p key="a" className="pwelcome2" > 
                  3.  Biodiversity and Ecosystem Services in Agricultural Landscapes
                  </p>
                  <p key="a" className="pwelcome3" > 
                  Many plant and animal species live in and/or use resources from agricultural landscapes.  We are interested in understanding which taxa live in these landscapes and how they use these landscapes.  We are especially interested in how biodiversity can provide ecosystem services within modified systems where we obtain much of our food from.  Our group studies biodiversity and ecosystem services across a variety of agricultural landscapes in Australia, including fruit orchards, grazing landscapes and field crops. This work is funded the UNE Postdoctoral Fellowship scheme.
                  </p> 
                  <p key="a" className="pwelcome2" > 
                  Our lab is based at the University of New England, Armidale NSW Australia.
                  </p>
            </Segment>   
          </div>
        </div>
      );
    }
  }

export  {PermissionPage};