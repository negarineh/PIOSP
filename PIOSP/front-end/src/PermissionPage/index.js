export * from './PermissionPage';
export * from './InhibitorPage';
export * from './NewUserAlertPage';
