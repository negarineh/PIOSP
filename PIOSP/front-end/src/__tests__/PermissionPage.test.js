import React from "react";
import Enzyme, { shallow } from "enzyme";
import assert from "assert";
import {PermissionPage} from "../PermissionPage";
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
 
Enzyme.configure({ adapter: new Adapter() });

// unit tests for the UserDeletePrompt component
describe('Permission Page component', () => {
  describe('render()', () => {
    it('should render the component', () => {
      const wrapper = shallow(<PermissionPage/>);
      const tree = toJson(wrapper);
      expect(tree).toMatchSnapshot();
      assert.equal(wrapper.length, 1);
    });
  });
});
