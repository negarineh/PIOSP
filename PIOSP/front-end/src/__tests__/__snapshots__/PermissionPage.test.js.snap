// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Permission Page component render() should render the component 1`] = `
<div
  className="mainpage-permission"
>
  <div
    className="butterfly"
  >
    <div
      className="body"
    >
      <img
        alt=""
        src="http://fjordstudio.dk/animations/img/butterfly3-body.png"
      />
    </div>
    <div
      className="leftwing"
    >
      <img
        alt=""
        src="http://fjordstudio.dk/animations/img/butterfly3-leftwing.png"
      />
    </div>
    <div
      className="rightwing"
    >
      <img
        alt=""
        src="http://fjordstudio.dk/animations/img/butterfly3-rightwing.png"
      />
    </div>
  </div>
  <div
    className="wrappermainpage-permission"
  >
    <Segment
      style={
        Object {
          "paddingLeft": "10px",
        }
      }
    >
      <p
        className="pwelcome1"
        key="a"
      >
        We study community ecology in agroecosystems
      </p>
      <p
        className="pwelcome2"
        key="a"
      >
        1.  Crop Pollinator Identity and Ecology
      </p>
      <p
        className="pwelcome3"
        key="a"
      >
        Pollinator communities include native and introduced bees, flies, beetles, moths, butterflies, and birds, as well as other taxa. These taxa provide pollination services for up to 87.5% of all flowering plants and 75% of the leading global food crops. We are interested in finding out which pollinators visit which crops in a range of agroecosystems. This project is supported by funding from the Australian Government Department of Agriculture and Water resources as part of its Rural R&D for Profit Programme and the Ian Potter Foundation.
      </p>
      <p
        className="pwelcome2"
        key="a"
      >
        2.  The Mechanisms Underlying Crop Pollinator Effectiveness
      </p>
      <p
        className="pwelcome3"
        key="a"
      >
        Maximally effective pollination results from a complex assortment of factors that influence the pollination process before and after pollen deposition. The effectiveness of individual pollinators is determined by the combined outcomes of the amount, quality and timing of the pollen transferred which is strongly related to pollinator foraging behaviour, morphology and density. Variation in the identity and spatial and temporal structuring of plant and pollinator communities further affect effective pollination, impacting pollen availability and post-pollination factors, such as pollen-pistil and/or pollen-pollen interactions on the stigma. Pollination failure can thus result from problems at any or all of these stages of pollen transfer. This project is funded by the Australian Research Council via a Discovery Early Career Researcher Award (DECRA) awarded to Romina Rader (DE170101349).
      </p>
      <p
        className="pwelcome2"
        key="a"
      >
        3.  Biodiversity and Ecosystem Services in Agricultural Landscapes
      </p>
      <p
        className="pwelcome3"
        key="a"
      >
        Many plant and animal species live in and/or use resources from agricultural landscapes.  We are interested in understanding which taxa live in these landscapes and how they use these landscapes.  We are especially interested in how biodiversity can provide ecosystem services within modified systems where we obtain much of our food from.  Our group studies biodiversity and ecosystem services across a variety of agricultural landscapes in Australia, including fruit orchards, grazing landscapes and field crops. This work is funded the UNE Postdoctoral Fellowship scheme.
      </p>
      <p
        className="pwelcome2"
        key="a"
      >
        Our lab is based at the University of New England, Armidale NSW Australia.
      </p>
    </Segment>
  </div>
</div>
`;
